src/movement/ThrusterController.cpp

```diff
@@ -0,0 +1,187 @@
+#include "movement/ThrusterController.hpp"
+namespace rs
+{
+    /**
+     * Constructor.
+     *
+     * @param thrusters The number of thrusters controlled by the controller.
+     * @param max_speed The maximum normalized speed in the forward and reverse
+     *        directions that should be allowed by the controller.
+     * @param delay_ms Delay specified in milliseconds to continue sending
+     *        zeroing signal to continue thruster operation.
+     * @param port The serial port to be used for communicating with the
+     *        thrusters. This port may be changed during runtime.
+     */
+    ThrusterController::ThrusterController(const int thrusters,
+                                           const double max_speed,
+                                           Serial *port,
+                                           const double delay_ms) :
```

Comment:
Constructor should be empty, pass in these parameters in an "init" function. This allows us to declare the class before we know how many thrusters there will be.